### DISCUSSION TITLE
Example Miner Hardware Profiles

### DISCUSSION THREAD

**Discussion by TippyFlitsUK** [filecoin-project/lotus#6071] [Show and tell]
### Server 1 - PC1, C2, Miner
**CPU** - Dual AMD EPYC 7282 - 16Core - 2.8GHz
**GPU** - Dual Nvidia Tesla T4 - 16GiB
**RAM** - 1TiB DDR4 3200 MHz ECC
**Sealing Media** - Dual 6.4TB Samsung NVMe
**Storage Medi**a - Seagate Exos 16TB Enterprise HDD
**Networking** - Intel X540-T2BLK Gigabit NIC
**Networking** - Broadcom N225P - 2 x 25/10GbE OCP 3.0 Adapter
**HBA** - 8 Port Broadcom SAS 9300-8e SGL, 12Gb/s SATA+SAS, PCIe 3.0 HBA
**External Storage Media** - SuperChassis 847E2C-R1K28JBOD - 44 Disk 4U
**Server** - Gigabyte R282-Z93 EPYC Server

### Server 2 - PC1, C2, Markets, Daemon, Backup
**CPU** - Dual AMD EPYC 7282 - 16Core - 2.8GHz
**GPU** - Nvidia RTX A4000 - 16GiB
**RAM** - 500GiB DDR4 3200 MHz ECC
**Sealing & Markets** -  6.4TB Samsung NVMe
**Storage Media** - Seagate Exos 16TB Enterprise HDD
**Networking** - Broadcom N225P - 2 x 25/10GbE OCP 3.0 Adapter
**HBA** - 8 Port Broadcom SAS 9300-8e SGL, 12Gb/s SATA+SAS, PCIe 3.0 HBA
**External Storage Media** - SuperChassis 847E2C-R1K28JBOD - 44 Disk 4U
**Server** - Gigabyte R282-Z93 EPYC Server

**Comment by TippyFlitsUK** [comment]
**Slack** - Kyle

**Server 1**
**CPU** - Threadripper 3960x
**RAM** - 32GiB - 2133 - Non ECC
**Sealing Media** - Local NVME

**Server 2**
**CPU** - Epyc 7351p
**RAM** - 8x64GiB - 2133 - ECC
**Sealing Media** - NFS NVME

**Server 3**
**CPU** - Ryzen 3900x
**RAM** - 32GB - 3200 - Non ECC
**Sealing Media** - Local NVME

**Reply by tutorialevideo** [reply]
is this config working well?

**Reply by djmartberry66** [reply]
Djsmartberry

**Comment by Reiers** [comment]
**TechHedge Home Mining**
**Slack** - @Reiers 

**Miner+Daemon:**
**CPU** - AMD® Ryzen Threadripper Pro 3995wx
**GPU** - NVIDIA TITAN RTX
**RAM** - 512GB 3200MHz DDR4 ECC
**Sealing Media** - WD BLACK 4TB NVMe
**Storage Media** - Seagate Exos 18TB Enterprise HDD
**Networking** - NETGEAR 10G Ethernet Plus Switch (XS708E)
**Server** - Storinator Q30 Turbo

**Storage Server:**
_Storinator Q30 Turbo, total units = 3
Seagate Exos 18TB Enterprise HDD, total disks = 90_
**CPU** - Dual Intel Xeon Silver 4210
**RAM** - 256GB

**Worker Cluster:**
**PC1 Workers** - PowerEdge R7515 Server
**CPU** - AMD EPYC 7F32
**RAM** - 1024GB 3200MHz ECC
**Sealing Media** - Samsung 860 PRO 2TB RAID 0, total disk per unit = 8

**Total PC1 Workers = 2**

**PC2 Workers** 
**CPU** - Intel Core i9-10900X _or similar_
**GPU** - NVIDIA RTX 3090
**RAM** - 256 GB 3200MHz + Swap 512GB
**Sealing Media** - WD BLACK 4TB NVMe

**Total PC2 Workers = 5**

**C2 Workers**
**CPU** - Intel Core i9-10940X _or similar_ 
**GPU** - NVIDIA RTX 3090
**RAM** - 256 GB 3200MHz + Swap 512GB
**Sealing Media** - WD BLACK 4TB NVMe

**Total C2 Workers = 6**

**Comment by TippyFlitsUK** [comment]
**Slack** - BigBear

**CPU** - Threadripper 3960x
**RAM** - 8x32GB - 3200 - non ECC
**Sealing Media** - Local NVME

**Reply by mxz9102** [reply]
Slack - BigBear
1.	AMD threadripper 3990x, Nvidia 2080ti GPU, 256GB memory, 24 hard drive bays
2.	AMD threadripper 3990x, Nvidia 2080ti GPU, 256GB memory, 24 hard drive bays
3.	AMD threadripper 3990x, 256GB memory, 24 hard drive bays
4.	AMD threadripper 3970x, 256GB memory, 24 hard drive bays
5.	AMD threadripper 3970x, 256GB memory, 5 hard drive bays
6.	Dell R series Intel XEON Processor 8 Core 1TB SSD RAID1 (Enterprise SSD)
7.	Lenovo laptop computer, 32GB memory, 1T SSD.
8.	Two Nvidia 3090 GPU.
9.	48 Seagate Exos enterprise 16TiB HDD

**Comment by kernelogic** [comment]
Slack - feiya200

Two identical PCs:
CPU - Threadripper 3960x
MB   - MSI TRX40 Pro 10G
RAM - 8x32GB - 3200 - non ECC
GPU - 2080 Ti
Sealing Media - SX8200 Pro NVME
Storage Media - a variety of shucked 16TB and 14TBs.

**Comment by fanfu13** [comment]
Slack - Frank 

Daemon
CPU - Intel(R) Xeon(R) W-3223 CPU
RAM - 128G 2600 DDR4 ECC
Storage Media:  3 * 2TB nvme to form a lvm 
Networking -  1G Ethernet 

Miner:
CPU - AMD 3960X
GPU - NVIDIA 2080Ti
RAM - 128GB 2600 DDR4 ECC
Sealing Media - 1 TB nvme
Storage Media - 8+3 EC ceph storage system
Networking -  2 * 10G Ethernet 

Storage Server:
FusionServer Pro 5288 V5, total units = 11
Seagate 14TB Enterprise HDD, total disks = 396
CPU - Dual Intel(R) Xeon(R) Silver 4114 CPU
RAM - 192GB
Networking -  2 * 10G Ethernet 

Worker Cluster:
PC1&P2 Workers 
CPU - AMD EPYC 7452
GPU - 2080Ti
RAM - 1024GB 2600MHz ECC 
Sealing Media - Samsung 7.84TB NVME, total disk per unit = 3
Networking -  10G Ethernet 

Total PC1&P2 Workers = 10

C2 Workers
CPU - Intel(R) Xeon(R) W-3245M CPU 
GPU - NVIDIA 2080Ti
RAM - 256 GB 
Sealing Media -512GB NVME
Networking -  1G Ethernet 

Total C2 Workers = 14

**Reply by Shekelme** [reply]
Why did you choose LVM? Isn't mdadm better in terms of performance (I read somewhere that LVM is working on top of mdadm)?

**Comment by stuberman** [comment]
Slack - @stuberman (driver levels updated April 2024)

**Miner+Daemon:**
OS - Ubuntu 20.04.3 server -> now Ubuntu 22.04.4 (April 2024)
CPU - AMD® Ryzen 9 3900X - 12 core/24 thread 3.8 GHz
RAM - Kingston KSM26ED8/32ME (4 x 32GB) DDR4 2666 (PC4 21300) ECC UDIMM
[ formerly CORSAIR Vengeance LPX 128GB (4 x 32GB) DDR4 2666 (PC4 21300) non-ECC]
GPU - NVIDIA 2080Ti - NVidia Driver 535.171.04 (Originally 1080Ti with 455.38 driver) - CUDA 12.2
OS and Lotus media: Sabrent Rocket 4 NVMe 4.0 Gen 4 PCIe: 1TB, 2TB
Motherboard - ASRock X570 PRO4 AM4 AMD X570
Case - SuperMicro SC846 BE2c-R1K28B (4x6 bays)
Storage Media - WD UltraStar HC530 14TB Enterprise HDD x 25 - 290 TiB usable, 230 TiB allocated for sector storage
Storage Pool - ZFS raidz - using 4 x 6 drive vdevs and one hot spare
HBA - LSI Broadcom SAS 9300-8i 8-port 12Gb/s SATA+SAS PCI-Express 3.0 

**Worker:**
OS - Ubuntu 22.04.4 server
CPU - AMD® Ryzen ThreadRipper 3990X - 64 core/128 thread
RAM - G.SKILL 256GB(8x32GB) F4-3200C16-32GTZN Trident Z Neo 3200 (PC4 25600) non-ECC
GPU - Nvidia 3090 - NVidia Driver 535.171.04 - CUDA 12.2 
OS and Lotus media: Sabrent Rocket 4 1TB NVMe 4.0 Gen 4 PCIe; and Sabrent 500GB NVMe
Sealing Media - Sabrent Rocket Q4 2x4TB NVMe 4.0 Gen 4 PCIe
Motherboard - ASUS ROG Zenith II Extreme Alpha TRX40 Gaming AMD 3rd Gen Ryzen Threadripper sTRX4 EATX 
Cooler - Noctua NH-U14S TR4-SP3 + Noctua NF-A15 PWM
PSU - EVGA Titanium 1600 Watt T2

**Market/Boost node:**
OS - Ubuntu 22.04.4 server
CPU - AMD® Ryzen 5 1600 6 core/12 thread
RAM - 64 GB G.Skill Flare X F4-2400C16-16GFX (4x16 2400 DDR4) non-ECC
GPU - Radeon HD 6450 - only used for console video
OS Disk - KingDian 2.5" S200 Sata III 60 GB SSD
Deal repo and storage 1x Sabrent Rocket 4TB NVMe PCIe Gen 3
Motherboard - x370 Gigabyte GA-AB350M-DS3H

**Battery backup/UPS:**
4x APC Smart-UPS SMT1000i (240volt) with USB connectivity to servers

**Internet:**
1 Gbps up/down - no cap - fiber line

**spare/test node:**
OS - Ubuntu 22.04.4 server
CPU - AMD® Ryzen 9 5900x 12 core/24 thread 3.7 GHz
RAM - Kingston KSM32ED8/32ME (4 x 32GB) DDR4 3200 (PC4 25600) ECC UDIMM
GPU - Nvidia GeForce GTX 1080Ti - NVidia Driver 535.171.04 - CUDA 12.2 
OS Disk - Sabrent Rocket 4 1 x 1TB NVMe 4.0 Gen 4 PCIe and Sabrent Rocket 4 SB-RKT4P-2TB 1 x 2TB NVMe 4.0 Gen 4 PCIe
Storage Media - WD UltraStar HC530 14TB Enterprise HDD x 8 - 72 TiB usable, 58 TiB allocated for sector storage
Storage Pool - ZFS raidz - using 1 x 7 drive vdev and one hot spare
Motherboard - ASRock X570 PRO4 AM4 AMD X570

**Reply by Imccccc** [reply]
how many sector does this cluster seal?

**Reply by sharli91** [reply]
Hello

 excuseme Sr. Could you explain how install file coin minner on Ubuntu server 20.04 

Reagars

**Reply by Shekelme** [reply]
> Hello
> 
> excuseme Sr. Could you explain how install file coin minner on Ubuntu server 20.04

[https://docs.filecoin.io/mine/](https://docs.filecoin.io/mine/)

**Comment by 6enno** [comment]
**Slack:** @Koru (Tom), @6enno (Ben)

**Miner + Daemon:**
CPU - Ryzen 3800x
GPU - 2080Ti
RAM - 128G 2933 (DDR4 non-buffered non ecc which may cause issues..)
Sealing Media - none
Storage Media - 8TB HDDs linked on backplane through HBA. not raid. Will move 
Networking - 10g eth pcie card linked to 10g switch.

**PC1 Box** (about 7 in parallel works well)
CPU - 2x EPYC7282
GPU - None
RAM - 640G DDR4 2666 ECC
Sealing Media - 5x 2TB NVMe in Raid0
Storage Media - None
Networking - 10g eth port linked to 10g switch

**PC2 Box**
CPU - EPYC 7402
GPU - 2080ti
RAM - 256 DDR4 2933 ECC (32GB each stick)
Sealing Media - 1x 2TB NVMe
Storage Media - None
Networking - 10g PCIE Eth card to 10g eth switch

**C2 Box**
CPU - EPYC 7232P
GPU - 3090
RAM - 384G DDR4 2666 ECC (64GB sticks)
Sealing Media - 1x 2TB NVMe
Storage Media - none
Networking - 10g Eth card to 10g switch

Note that this all gets chopped and changed pretty often. Looking to raidz storage and also try sealing on raid0 SSDs in the near future.

**Comment by neondragon** [comment]
**Neonix**
**Slack: @NeonixAF f019551** (Alex)

**Miner + Daemon (Dell Precision R7910, 2U)**
CPU - 2x Intel Xeon E5-2680 v3
GPU - 1x NVIDIA Titan V
RAM - 256GB (8x32GB) 2133MHz ECC DDR4
Local Storage - 6x 1.6TB Intel DC S3510, ZFS mirror, 4.8TB usable
Filecoin Sector Storage - 2+1 EC Ceph storage cluster (3 machines) with 16TB Seagate Exos drives
Networking - Dual port 10GbE card

**Storage Servers (Ceph cluster, 6U)**
3x 2u servers each with 2x Xeon E5-2670, 128GB RAM, 8x 16TB Seagate Exos drives, Dual-port 10GbE card
Layout is currently 2+1 erasure code. Intending to add more machines to the cluster soon.

**PC1 worker (2U)**
CPU - 2x AMD EPYC 7551
GPU - None
RAM - 1TB DDR4 (16x64GB) 2667MHz ECC DDR4
OS Storage - 256GB NVMe
Seal Storage - 2x 4TB NVMe, 2x 2TB NVMe, striped with LVM
Proofs/Parents cache storage - 1x 2TB NVMe
Networking - Dual port 10GbE card

**PC2/C2 worker 1 (4U)**
CPU - AMD EPYC 7262
RAM - 256GB (8x32GB) 2133MHz ECC DDR4
GPU - 1x NVIDIA RTX 3090, 1x NVIDIA RTX 3080
Seal Storage - NFS mounted storage from PC1 worker
OS/Proofs/Parents cache storage/swap - 2x 1TB NVMe (MD RAID1)
Networking - Dual-port 10GbE card

**PC2/C2 worker 2 (4U)**
CPU - AMD Ryzen 3900X
RAM - 128GB (4x32GB) 3200MHz DDR4
GPU - 1x NVIDIA RTX 3080
Seal Storage - NFS mounted storage from PC1 worker
OS Storage - 2x 250GB SSD (MD RAID1)
Proofs/Parents cache storage/swap - 1x 500GB NVMe
Networking - Dual-port 10GbE card

All machines connected to an APC UPS (4U), and 2x 10GbE switches (2U).

**Comment by flyworker** [comment]
Slack: @Charles Cao- Filswan
Miner + Daemon (HPE 385, 2U)
CPU -  Intel Xeon 4210R
GPU - 1x 2080TI
RAM - 256GB (8x32GB) 2667MHz ECC DDR4
Local Storage - 4 * 2TB WD SSD 
Networking - Dual port 10GbE card

Storage Servers (Ceph cluster, 6U)
3x 2u servers each with 2x Xeon E5-2670, 128GB RAM, 8x 16TB Seagate Exos drives, Dual-port 10GbE card

PC1/P2/C2 worker (4U)
CPU -  AMD EPYC 7402
GPU - 2x 2080TI
RAM - 1TB DDR4 (16x64GB) 2667MHz ECC DDR4
OS Storage - 1TB NVMe
Seal Storage - 4x 4TB NVMe, striped with LVM
Networking - Dual port 10GbE card

**Reply by Shekelme** [reply]
Why LVM, Charles? I heard that LVM is running on top of mdadm - isn't it better to use mdadm itself?

**Comment by Byeager1** [comment]
PC1 Machine | Quantity X 1
-- | --
TS2-185671979 - 2U Tackmount Server, Supports 1x EPYC ROME/Milan Processor, 8x DDR4 Modules, 6x 3.5" SATA Bays, 2x NVME U.2 Bays, 4x PCIE Gen 4 x 16 Slots, 2x PCIE Gen 4 LP Slots, Redundant Power Supplies | 1
EPYC 7F52 Processor - 3.5GHz - 16 Cores | 1
128GB DDR4 3200 ECC REG Memory (1TB Total RAM) | 8
3.84TB SATA 6Gb/s, 3D, TLC 2.5", 7.0mm SSD | 1
Micron MAX 9300 3.2 TB U.2 2.5", 7.0mm SSD | 4
Intel Quad-Port 10GbE (RJ45) | 1

PC2 Machine X 2 | Quantity
-- | --
Exxact TS2-185671979 - 1x EPYC 7320P, 32GBx8 DDR4, 1x 3.84TB SATA SSD, 4x 3.2TB NVMe SSD, 1x Quad-Port 10GBase-T NIC, 1z RTX 3080 GPU |  
EPYC 7320P - 16-Core - 3.0 GHz - Socket SP3 Processor | 1
32GB DDR4 3200 REG ECC Memory (256GB Total RAM) | 8
3.84TB, SATA 6Gb/s, 3D, TLC 2.5", 7.0mm SSD | 1
Micron MAX 9300 3.2 TB U.2 2.5" NVMe SSD | 4
NVIDIA RTX 3080, 8704 CUDA Cores, 10GB GDDR6X Memory, PCIe 4.0 GPU | 1
Intel Quad-Port 10GbE (RJ45) | 1


C1/C2 Machine X 2 |  
-- | --
Exxact TS2-185671979 - 1x EPYC 7320P, 64GBx8 DDR4, 4x 3.2TB NVMe SSD, 1x Quad-Port 10GBase-T NIC, 1x RTX 3090 24GB GPU |  
EPYC 7302P - 16-Core - 3.0 GHz - Socket SP3 Processor | 1
64GB DDR4 3200MHz REG ECC Memory Module | 8
Micron MAX 9300 3.2 TB U.2 2.5" NVMe SSD | 4
Nvidia RTX 3090, 10496 CUDA Cores, 24 GB GDDR6X Memory, PCIe 4.0 GPU | 1
Intel Quad-Port 10GbE (RJ45) | 1


S45 Storinator 45 Bay Storage Server X 2 |  
-- | --
AMD EPYC 7281 - 16 Core, 32 Threads - 2.10 GHz | 1
Motherboard: Supermicro H11SSL-I | 1
Power Supply: 1200W 2N Redundant Zippy Power Supply | 1
RAM: 256GB | 1
250GB SSD Boot Drive | 2
Chassis: 31.5"L x 17.125" W x 7.0"H \| Hot Swap 3.5" bays | 1
Seagate Hard Drives: Exos 18TB (ST18000NM000J) | 45
Dual Port 10 Gb/s NIC | 1
Add on NIC Dual Port 10Gb/s Copper | 2

**Reply by de-authority** [reply]
Nice bro.  the new miner will know how to buy hardware refer to your configuration,

**Reply by Imccccc** [reply]
how many sectors do your cluster seal per day?

**Reply by dcshy** [reply]
c2是否可使用远程

**Reply by de-authority** [reply]
> c2是否可使用远程
可以的

**Reply by Wondertan** [reply]
@Byeager1, can you pls share the PC1 time you get on 7F52?

**Reply by Shekelme** [reply]
How do you achieve the simultaneous work of all four ports of network cards? In my case, double-port cards use only one channel bandwidth ...

**Reply by Wondertan** [reply]
@Shekelme, search for "bonding network interfaces", e.g. https://wiki.debian.org/Bonding

There are multiple ways for doing bonding. Personally, I use 802.3ad

**Comment by johnwick147** [comment]
or you could just get a VPS and mine storx instead
